File: src/GPSLocation.cpp
//
// Created by Hazhir on 12/22/2023.
//

#include "GPSLocation.h"

float GPSLocation::getLatitude() const {
    return latitude;
}

void GPSLocation::setLatitude(float newLatitude) {
    GPSLocation::latitude = newLatitude;
}

float GPSLocation::getLongitude() const {
    return longitude;
}

void GPSLocation::setLongitude(float newLongitude) {
    GPSLocation::longitude = newLongitude;
}

GPSLocation::GPSLocation(float latitude, float longitude) : latitude(latitude), longitude(longitude) {}

File: src/dps/GPSLocation.java
package dps;
public class GPSLocation {
    double latitude;
    double longitude;

    public GPSLocation(double latitude, double longitude){
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    @Override
    public String toString() {
        return String.format("%f,%f", this.latitude, this.longitude);
    }

    public static GPSLocation fromString(String s) {
        String[] location = s.split(",");
        return new GPSLocation(Double.valueOf(location[0]), Double.valueOf(location[1]));
    }
}